### PR TITLE
Standardize transform identity calculation

### PR DESCRIPTION
**Pull request by lptr**
We've introduced a way to capture inputs for work items that are required to determine their identity separately from its regular inputs. This was added to enable running units of work through the execution engine that didn't have an identity assigned to them up-front (like tasks do). Some of the existing units of work already use this new approach, but transforms have been doing their own snapshotting and fingerprinting of some of their inputs because of historical reasons. This PR fixes that, and introduces a unified approach to capture identity vs regular inputs.

### CODE REVIEW

**Comment by lptr** [comment]
@bot-gradle test QF please.

File: subprojects/execution/src/main/java/org/gradle/internal/execution/UnitOfWork.java [hunk 1]
```diff
@@ -109,23 +111,25 @@ default void visitImplementations(ImplementationVisitor visitor) {
     }
 
     /**
-     * Visit all inputs of the work.
+     * Visit regular inputs of the work.
```
Comment (wolfs) [inline subprojects/execution/src/main/java/org/gradle/internal/execution/UnitOfWork.java:114]:
```suggestion
     * Visit identity inputs of the work.
```

File: subprojects/execution/src/main/java/org/gradle/internal/execution/UnitOfWork.java [hunk 2]
```diff
@@ -109,23 +111,25 @@ default void visitImplementations(ImplementationVisitor visitor) {
...
      */
-    void visitInputs(InputVisitor visitor);
+    default void visitIdentityInputs(InputVisitor visitor) {}
+
+    /**
+     * Visit regular inputs of the work.
```
Comment (wolfs) [inline subprojects/execution/src/main/java/org/gradle/internal/execution/UnitOfWork.java:119]: Can you clarify what "regular" inputs and "identity" inputs mean?

**Review by wolfs — APPROVED** [review]
I think the changes looks good. I left two minor comments.

File: subprojects/execution/src/main/java/org/gradle/internal/execution/UnitOfWork.java (hunk 1 above)
Comment (lptr) [reply subprojects/execution/src/main/java/org/gradle/internal/execution/UnitOfWork.java:114]: Fixed.

File: subprojects/execution/src/main/java/org/gradle/internal/execution/UnitOfWork.java (hunk 2 above)
Comment (lptr) [reply subprojects/execution/src/main/java/org/gradle/internal/execution/UnitOfWork.java:124]: Updated the Javadoc.

File: subprojects/execution/src/main/java/org/gradle/internal/execution/UnitOfWork.java [hunk 3]
```diff
@@ -111,12 +111,19 @@ default void visitImplementations(ImplementationVisitor visitor) {
...
+     * These are more expensive to calculate than regular inputs as they need to be calculated even if the execution of the work is short circuited by an identity cache.
+     * They also cannot reuse snapshots taken during previous executions.
+     * Because of these reasons only capture inputs as identity if they are actually used to calculate the identity of the work.
      */
     default void visitIdentityInputs(InputVisitor visitor) {}
 
     /**
      * Visit regular inputs of the work.
+     *
+     * Regular inputs are inputs that are not used to calculate the identity of the work, but used to check up-to-dateness or to calculate the cache key.
```
Comment (wolfs) [inline subprojects/execution/src/main/java/org/gradle/internal/execution/UnitOfWork.java:126]: Can you clarify if one need to visit inputs which are part of the identity input as well here if they are regular inputs? Maybe this method should be called `visitInputs` only.

**Review by wolfs — COMMENTED** [review]
I left another comment.

**Comment by lptr** [comment]
@bot-gradle test this please.

File: subprojects/execution/src/main/java/org/gradle/internal/execution/UnitOfWork.java (hunk 3 above)
Comment (lptr) [reply subprojects/execution/src/main/java/org/gradle/internal/execution/UnitOfWork.java:126]: I clarified.

**Comment by lptr** [comment]
@bot-gradle test this please.

**Comment by lptr** [comment]
@bot-gradle test this please.

**Comment by lptr** [comment]
@bot-gradle test this please.

**Pull request merged**